opentelemetry-prometheus/src/lib.rs

```diff
@@ -281,7 +273,10 @@ impl prometheus::core::Collector for Collector {
         let mut inner = match self.inner.lock() {
             Ok(guard) => guard,
             Err(err) => {
-                global::handle_error(err);
+                otel_error!(
```

Comment:
not sure of the impact of this error? is that going to fail entire metrics scraping? if yes, please use a more descriptive message. If not, use a lower severity. Either way, lets try to make sure the internal logs are descriptive enough.

Reply:
im not sure of the impact in these places, but i tried to update description and change two of them to warns :)

Reply:
Looks like you haven't pushed the commit? Not seeing the change!

Reply:
i've changed otel error to otel warning in validate metrics, here i think should be error as it results in empty vec.
here i changed only name, end error as a message

Reply:
> OpenTelemetryPrometheus.AcquireLockError

This name is not end user friendly. Could you use something like MetricScrapeFailed or something similar, avoiding any internal details like locks.

Reply:
also, there is no need to prefix OpenTelemetryPrometheus - it'll be automatically added as target!

Reply:
k, changed :)